examples/server/server.cpp

```diff
@@ -504,95 +500,207 @@ bool server_params_parse(int argc, char **argv, server_params &sparams, gpt_para
+    llama.params.repeat_last_n = body["repeat_last_n"].get<int>();
+  } else {
+    llama.params.repeat_last_n = default_params.repeat_last_n;
+  }
+  if (!body["temperature"].is_null()) {
     llama.params.temp = body["temperature"].get<float>();
+  } else {
+    llama.params.temp = default_params.temp;
   }
-  if (!body["batch_size"].is_null())
-  {
-    llama.params.n_batch = body["batch_size"].get<int>();
+  if (!body["repeat_penalty"].is_null()) {
+    llama.params.repeat_penalty = body["repeat_penalty"].get<float>();
+  } else {
+    llama.params.repeat_penalty = default_params.repeat_penalty;
   }
-  if (!body["n_keep"].is_null())
-  {
+  if (!body["presence_penalty"].is_null()) {
+    llama.params.presence_penalty = body["presence_penalty"].get<float>();
+  } else {
+    llama.params.presence_penalty = default_params.presence_penalty;
+  }
+  if (!body["frequency_penalty"].is_null()) {
+    llama.params.frequency_penalty = body["frequency_penalty"].get<float>();
+  } else {
+    llama.params.frequency_penalty = default_params.frequency_penalty;
+  }
+  if (!body["mirostat"].is_null()) {
+    llama.params.mirostat = body["mirostat"].get<float>();
+  } else {
+    llama.params.mirostat = default_params.mirostat;
+  }
+  if (!body["mirostat_tau"].is_null()) {
+    llama.params.mirostat_tau = body["mirostat_tau"].get<float>();
+  } else {
+    llama.params.mirostat_tau = default_params.mirostat_tau;
+  }
+  if (!body["mirostat_eta"].is_null()) {
+    llama.params.mirostat_eta = body["mirostat_eta"].get<float>();
+  } else {
+    llama.params.mirostat_eta = default_params.mirostat_eta;
+  }
+  if (!body["penalize_nl"].is_null()) {
+    llama.params.penalize_nl = body["penalize_nl"].get<float>();
+  } else {
+    llama.params.penalize_nl = default_params.penalize_nl;
+  }
+  if (!body["n_keep"].is_null()) {
     llama.params.n_keep = body["n_keep"].get<int>();
+  } else {
+    llama.params.n_keep = default_params.n_keep;
   }
-  if (!body["as_loop"].is_null())
-  {
-    llama.as_loop = body["as_loop"].get<bool>();
+  if (!body["seed"].is_null()) {
+    llama.params.seed = body["seed"].get<int>();
+  } else {
+    llama.params.seed = time(NULL);
   }
-  if (!body["interactive"].is_null())
-  {
-    llama.params.interactive = body["interactive"].get<bool>();
+
+  llama.params.logit_bias.clear();
+  if (!body["ignore_eos"].is_null() && body["ignore_eos"].get<bool>()) {
+    llama.params.logit_bias[llama_token_eos()] = -INFINITY;
   }
-  if (!body["prompt"].is_null())
-  {
-    llama.params.prompt = body["prompt"].get<std::string>();
+  if (body["logit_bias"].is_array()) {
+    int n_vocab = llama_n_vocab(llama.ctx);
+    for (const auto &el : body["logit_bias"]) {
+      if (el.is_array() && el.size() == 2 && el[0].is_number_integer() && el[1].is_number_float()) {
+        llama_token tok = el[0].get<llama_token>();
+        if (tok < 0 || tok >= n_vocab) continue;
+        llama.params.logit_bias[tok] = el[1].get<float>();
+      }
```

Comment:
```suggestion
      if (el.is_array() && el.size() == 2 && el[0].is_number_integer()) {
        llama_token tok = el[0].get<llama_token>();
        if (tok >= 0 && tok < n_vocab) {
          if (el[1].is_number_float()) {
            llama.params.logit_bias[tok] = el[1].get<float>();
          } else if (el[1].is_boolean() && !el[1].get<bool>()) {
            llama.params.logit_bias[tok] = -INFINITY;
          }
        }
      }
```

Since JSON doesn't have Infinity I'm using `false` here, `null` doesn't seem to work.